services/crawler/http.go

```diff
@@ -0,0 +1,50 @@
+package crawler
+
+import (
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+
+	"github.com/beego/beego/v2/core/logs"
+	"github.com/junan/fake-useragent"
+)
+
+func GetRequest(url string) ([]byte, error) {
+	client := &http.Client{}
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		logs.Error("Building new request failed: ", err)
+	}
+
+	req.Header.Set("User-Agent", RandomUserAgent())
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	// Closing response body
+	defer res.Body.Close()
+	byte, err := ioutil.ReadAll(res.Body)
+
+	return byte, err
+}
+
+// Returns random Chrome or Firefox user agent.
+// Browser version, system-information, platform also random.
+func RandomUserAgent() string {
+	randomNumber := GenerateRandomNumber()
+	switch randomNumber {
+	case 0:
+		return browser.Chrome()
+	case 1:
+		return browser.Firefox()
+	default:
+		return browser.Chrome()
+	}
+}
+
+func GenerateRandomNumber() int {
+	rangeLower := 0
+	rangeUpper := 1
+	return rangeLower + rand.Intn(rangeUpper-rangeLower+1)
```

Comment:
This logic seems complex to randomize between 0 and 1 🙃 ANy alternative?

Reply:
I also found it is a bit complex, but I have not found any other easy way to do it.
https://stackoverflow.com/questions/23577091/generating-random-numbers-over-a-range-in-go

Reply:
Hoang found out that Colly does have a random user-agent feature (see [code](https://github.com/gocolly/colly/blob/master/extensions/random_user_agent.go)). So maybe using it over a custom rolled-out solution would be better. What do you think?

Reply:
I am using [fake-useragent](https://github.com/eddycjy/fake-useragent), but I had to fork it, update with newer chrome and Firefox browser versions. Then, using that [forked version](https://github.com/junan/google-scraper/pull/29/files#diff-703ba1af48162940c9a2db65ac0bf3b2a63c1cae8379f7b181afd7587b92e804R9) . 

The reason for forking/updating is that, I found that Google gives very different DOM/Html response for older browser versions(Chrome less that 49 version and firefox less then 53, and for many versions for Opera) user agent and then the Html parser fails(cant extract proper data).

 I am not using Colly but it will not also work for me as it is using [older versions](https://github.com/gocolly/colly/blob/master/extensions/random_user_agent.go#L36) also for random agents. What is your thought?

Reply:
Indeed, you are not using Colly 😅 Still, inspiring yourself on how their randomization is done could be an idea 💭